### PR TITLE
Make rocksdb configurable

### PR DESCRIPTION
**Pull request by evgeniy-scherbina**
Related commit: https://github.com/Kava-Labs/tm-db/commit/7fb308e76593bafc0bb28e0b7fca058520e4d56d

Steps before merge:
- make `tm-db` release and change version in `go.mod` - `DONE`
- remove temporary `test.sh` script - `DONE`

Future improvements:
- somehow integrate `CheckOptionsCompatibility`
- add e2e tests?

### CODE REVIEW

File: .github/workflows/ci-rocksdb-build.yml [hunk 1]
```diff
@@ -17,6 +17,6 @@ jobs:
       - name: build rocksdb dependency
         run: bash ${GITHUB_WORKSPACE}/.github/scripts/install-rocksdb.sh
         env:
-          ROCKSDB_VERSION: v7.10.2
+          ROCKSDB_VERSION: v8.1.1
```
Comment (galxy25) [inline .github/workflows/ci-rocksdb-build.yml:20]: https://github.com/Kava-Labs/infrastructure/blob/master/packer/kava-ami/kava-internal-testnet-rocksdb.pkrvars.hcl is what internal testnet is using, will need to update the config and update existing nodes using the kava-node-updater(see https://kava-labs.atlassian.net/wiki/spaces/ENG/pages/1105231874/In+Place+Updates+of+a+Kava+Network and https://kava-labs.atlassian.net/wiki/spaces/ENG/pages/1147174913/How+to+upgrade+Kava+Mainnet+or+Testnets#Kava-Node-Updater-CLI for details - for everywhere this change needs to be deployed to (e.g. https://github.com/Kava-Labs/infrastructure/blob/master/packer/kava-ami/kava-public-testnet-rocksdb.pkrvars.hcl and https://github.com/Kava-Labs/infrastructure/blob/master/packer/kava-ami/kava-14-rocksdb.pkrvars.hcl) - doesn't need to be done in this pr but @evgeniy-scherbina please add a shortcut ticket to the epic for this work to track it

**Review by galxy25 — APPROVED** [review]

File: cmd/kava/opendb/opendb_rocksdb.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+		return nil, nil, ErrUnexpectedConfiguration
+	}
+
+	// return db and cf opts
+	return latestOpts.Options(), &cfOpts[0], nil
+}
+
+// overrideDBOpts merges dbOpts and appOpts, appOpts takes precedence
+func overrideDBOpts(dbOpts *grocksdb.Options, appOpts types.AppOptions) *grocksdb.Options {
+	maxOpenFiles := appOpts.Get(maxOpenFilesDBOptName)
```
Comment (nddeluca) [inline cmd/kava/opendb/opendb_rocksdb.go:101]: These maybe should be prefixed so they can be grouped in app.toml? `db.max_open_files`, `rocksdb.max_open_files`, etc? Not sure the best prefix

**Review by nddeluca — APPROVED** [review]
Looks good.  We will have quite a few more options in the future, so we'll likely want to group by a prefix at some point for better organization & a unique namespace

**Review by pirtleshell — APPROVED** [review]

File: cmd/kava/opendb/opendb_rocksdb.go (hunk 2 above)
Comment (evgeniy-scherbina) [reply cmd/kava/opendb/opendb_rocksdb.go:101]: `rocksdb.max_open_files` looks good

**Pull request merged**